packages/utils/src/builders/embeds.ts

```diff
@@ -0,0 +1,387 @@
+   */
+  validate(): EmbedsBuilder {
+    let totalCharacters = 0;
+
+    if (this.length > 10) {
+      throw new EmbedsBuilderError('You can not have more than 10 embeds on a single message.')
+    }
+
+    this.forEach(({ author, description, fields, footer, title}, index) => {
+      if (title) {
+        const trimmedTitle = title.trim()
+
+        if (trimmedTitle.length > 256) {
+          throw new EmbedsBuilderError(`Title of embed ${index} can not be longer than 256 characters.`)
+        }
+
+        totalCharacters += trimmedTitle.length
+      }
+
+      if (description) {
+        const trimmedDescription = description.trim()
+
+        if (trimmedDescription.length > 4096) {
+          throw new EmbedsBuilderError(`Description of embed ${index} can not be longer than 4096 characters.`)
+        }
+
+        totalCharacters += trimmedDescription.length
+      }
+
+      if (fields) {
+        if (fields.length > 25) {
+          throw new EmbedsBuilderError(`embed ${index} can not have more than 25 fields.`)
+        }
+
+        fields.forEach(({ name, value }, fieldIndex) => {
+          const trimmedName = name.trim()
+          const trimmedValue = value.trim()
+
+          if (trimmedName.length > 256) {
+            throw new EmbedsBuilderError(`Name of field ${fieldIndex} on embed ${index} can not be longer than 256 characters.`)
+          }
+
+          if (trimmedValue.length > 4096) {
+            throw new EmbedsBuilderError(`Value of field ${fieldIndex} on embed ${index} can not be longer than 1024 characters.`)
+          }
+
+          totalCharacters += trimmedName.length
+          totalCharacters += trimmedValue.length
+        })
+      }
+
+      if (footer) { 
+        const trimmedFooterText = footer.text.trim()
+
+        if (trimmedFooterText.length > 2048) {
+          throw new EmbedsBuilderError(`Footer text of embed ${index} can not be longer than 2048 characters.`)
+        }
+
+        totalCharacters += trimmedFooterText.length
+      }
+
+      if (author) {
+        const trimmedAuthorName = author.name.trim()
+
+        if (trimmedAuthorName.length > 256) {
+          throw new EmbedsBuilderError(`Author name of embed ${index} can not be longer than 256 characters.`)
+        }
+
+        totalCharacters += trimmedAuthorName.length
+      }
+    })
+
+    if (totalCharacters > 6000) {
+      throw new EmbedsBuilderError('Total character length of all embeds can not exceed 6000 characters.')
+    }
+
+    return this
+  }
+  
+  /**
+   * Returns the current embed.
+   *
+   * @readonly
+   * @type {DiscordEmbed}
+   */
+  get #currentEmbed(): DiscordEmbed {
+    if (this.length === 0) {
+      this.newEmbed()
+      this.setCurrentEmbed()
+    }
+ 
+    return this[this.#currentEmbedIndex]
+  }
+}
+
+/**
+ * Custom EmbedsBuilder error.
+ *
+ * @class EmbedsBuilderError
+ * @typedef {EmbedsBuilderError}
+ * @extends {Error}
+ */
+class EmbedsBuilderError extends Error {
```

Comment:
Why is this class needed?

Reply:
@Skillz4Killz mainly just experience. Setting a custom error allows you to check, if desired, what the error name is and allows you to handle that error differently than others.

There's no real downside to not throwing generic errors vs custom errors. I don't feel strongly either way, I can just throw a generic error instead if that's what we decide.

Reply:
I think we should keep it generic please. You can compare error message as well.

Reply:
Resolved in 38e1a5ba0678c5db45cb117a605da37903fd6dc1